.github/workflows/fixup.yml

```diff
@@ -0,0 +1,28 @@
+name: 'Commit Message Check'
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    branches:
+      - main
+
+jobs:
+  check-commit-message:
+    name: Check Commit Message
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check Fixup
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^(?!.*FIXUP).*'
```

Comment:
As @treiher pointed out in our last discussion about this we may want to check for `WIP` and `no-ci`, too. Additionally this regex only finds upper case fixups (which for example would not have prevented my fixup commit).
My suggestion would be
```
^(?!.*(?i)(fixup|wip|no.?ci)).*
```
as this matches any case of `fixup`, `wip` and `no-ci` (with and without a character between `no` and `ci`).

Reply:
I agree regarding WIP, but matching it as you suggest is dangerous as "wip" may be part of legit commit messages. I added it and changed it to match both, WIP and FIXUP as whole words only.

The matching is case-insensitive already (that's what the `i` in `flags: 'gi'` is for). The `(?i)` construct you suggest is not available in JavaScript.

I'm not convinced that we should prevent `no ci`. Firstly, it's much [more complicated than you might think](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs). Secondly, we enforce a successful CI run - why should suppressing the CI in some intermediate commit be a problem?

Reply:
> The matching is case-insensitive already (that's what the i in flags: 'gi' is for). The (?i) construct you suggest is not available in JavaScript.

I see, I wasn't aware of that.

> I'm not convinced that we should prevent no ci. Firstly, it's much [more complicated than you might think](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs). Secondly, we enforce a successful CI run - why should suppressing the CI in some intermediate commit be a problem?

Good point.